pkg/registryclient/client.go

```diff
@@ -36,7 +37,7 @@ func Initialize(client kubernetes.Interface, namespace, serviceAccount string, i
 		return errors.Wrap(err, "failed to initialize registry keychain")
 	}
 
-	authn.DefaultKeychain = kc
+	DefaultKeychain = kc
```

Comment:
Nice. I don't know if this is a concern, but this means the job will lookup secrets once at init-time, and use those secrets for the lifetime of the job. If new secrets are added (to the SA or as imagepullsecrets on the pod) those won't be used until the next time initialize picks them up.

But this is undeniably an improvement, so lgtm!

Reply:
@imjasonh I believe it should update it each time https://github.com/kyverno/kyverno/blob/7a8c19e0cbccb1f6cee9f77d8d017dd07680f0f9/pkg/registryclient/client.go#L44 is called

Reply:
Oh excellent! Thanks! 👍